Source/WebCore/dom/FullscreenManager.h

```diff
@@ -122,6 +128,8 @@ class FullscreenManager final : public CanMakeWeakPtr<FullscreenManager>, public
     Deque<GCReachableRef<Node>> m_fullscreenChangeEventTargetQueue;
     Deque<GCReachableRef<Node>> m_fullscreenErrorEventTargetQueue;
 
+    WeakHashMap<Document, OptionSet<ResizeType>, WeakPtrImplWithEventTargetData> m_documentsPendingScheduledResize;
```

Comment:
A Fullscreen manager has a 1:1 relationship with the Document that owns it. So a WeakHashMap would be overkill here; it would only ever have a single Document in it. 

You should be able to simply make an OptionSet ivar here and use it in setAnimatingFullscreen() to determine if either event should be fired.

Reply:
Done.